File: scratch/detect_errors.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  
  const errors = [];
  page.on('pageerror', err => errors.push(err.message));
  page.on('console', msg => {
    if (msg.type() === 'error') errors.push(msg.text());
  });

  try {
    await page.goto('http://localhost:8080', { waitUntil: 'networkidle', timeout: 5000 });
    // Wait a bit for the game to run a few frames
    await page.waitForTimeout(2000);
  } catch (e) {
    console.log('Timeout/Load error:', e.message);
  }

  if (errors.length > 0) {
    console.log('--- CONSOLE ERRORS FOUND ---');
    errors.forEach(e => console.log(e));
    console.log('--- END ERRORS ---');
  } else {
    console.log('No console errors detected.');
  }

  await browser.close();
  process.exit(0);
})();

File: scratch/detect_errors.py
import asyncio
from playwright.async_api import async_playwright

async def run():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()
        
        errors = []
        page.on("pageerror", lambda exc: errors.append(f"PAGE ERROR: {exc}"))
        page.on("console", lambda msg: errors.append(f"CONSOLE ERROR: {msg.text}") if msg.type == "error" else None)

        print("Visiting game...")
        try:
            await page.goto("http://localhost:8080", wait_until="networkidle", timeout=5000)
            await asyncio.sleep(2)  # Wait for frames
        except Exception as e:
            print(f"Navigation failed: {e}")

        if errors:
            print("--- CONSOLE ERRORS FOUND ---")
            for err in errors:
                print(err)
            print("--- END ERRORS ---")
        else:
            print("No console errors detected.")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(run())
